app/controllers/SicAndComplianceController.scala

```diff
@@ -128,14 +133,21 @@ class SicAndComplianceController @Inject()(val authConnector: AuthClientConnecto
           iclCodes <- iclService.getICLSICCodes()
           _ <- keystoreConnector.cache(SIC_CODES_KEY, iclCodes)
           _ <- sicAndCompService.submitSicCodes(iclCodes)
+          vatScheme <- vatRegistrationService.getVatScheme
+          isSoleTrader = vatScheme.eligibilitySubmissionData.exists(_.partyType.equals(Individual))
         } yield {
           Redirect(iclCodes match {
             case codes if codes.size > 1 =>
               routes.SicAndComplianceController.showMainBusinessActivity()
             case codes if sicAndCompService.needComplianceQuestions(codes) =>
               controllers.routes.SicAndComplianceController.showComplianceIntro()
             case List(onlyOneCode) =>
-              controllers.registration.business.routes.TradingNameController.show()
```

Comment:
I was thinking you could redirect this to a new controller that does the isSoleTrader logic and redirects accordingly. That way you don't need to duplicate
          vatScheme <- vatRegistrationService.getVatScheme
          isSoleTrader = vatScheme.eligibilitySubmissionData.exists(_.partyType.equals(Individual))
everywhere, or you could create an isSoleTrader def in the VatRegistrationService to offload the check there but keep the redirects in every relevant controller